### PR TITLE
EVA-2553 - Release Record writer and tests

### PR DESCRIPTION
**Pull request by sundarvenkata-EBI**
Implementation of [Incremental release strategy](https://docs.google.com/spreadsheets/d/1D3ipb1ELjyzGUEoOtlDsEsUHujyfCtPlc0YUvu37eNM/edit#gid=0) via release records. Usage [here](https://www.ebi.ac.uk/panda/jira/browse/EVA-2553?focusedCommentId=369092&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-369092).

### CODE REVIEW

**Review by tcezard — COMMENTED** [review]
I will leave others to comment on the code but I have been trying to understand how this could be run in the normal submission context. These comments are only here to plan how we can use this strategy outside of the covid19 context.

It needs to run after the accessioning (obviously), the possible remapping and the clustering.
But it starts from the accessioning report as far as I can tell. 
It also seem to crash if the encountered ssid has not been clustered which might happen if the remapping was unsuccessful. 
I also think it will be complex to find the correct accessioning report that needs to be used.

**Review by apriltuesday — APPROVED** [review]
Looks good, I learned a lot from reviewing this too :+1:

File: eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/batch/io/ReleaseRecordWriter.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+                                                                Collections.singletonList(submittedVariantEntity))));
+            }
+        }
+        return releaseInfo;
+    }
+
+    /**
+     * Get both SS and RS information (termed as a "release record") given a set of SS IDs
+     * @param ssIds - List of SS IDs
+     * @return A dictionary of RS and its constituent SS - keyed by the RS ID
```
[the remark this answers is not in the document]
Comment (sundarvenkata-EBI) [reply eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/batch/io/ReleaseRecordWriter.java:155]: Will address.

File: eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/configuration/batch/steps/CreateIncrementalReleaseStepConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import uk.ac.ebi.eva.accession.release.configuration.batch.processors.ReleaseProcessorConfiguration;
+import uk.ac.ebi.eva.accession.release.parameters.InputParameters;
+import uk.ac.ebi.eva.commons.batch.io.UnwindingItemStreamReader;
+import uk.ac.ebi.eva.commons.batch.io.VcfReader;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+
+import java.io.File;
+import java.io.IOException;
+
+import static uk.ac.ebi.eva.accession.release.configuration.BeanNames.*;
```
[the remark this answers is not in the document]
Comment (sundarvenkata-EBI) [reply eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/configuration/batch/steps/CreateIncrementalReleaseStepConfiguration.java:47]: Will address.

**Pull request merged**